src/coreComponents/fileIO/python/PyVTKOutput.cpp

```diff
@@ -0,0 +1,204 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+// Source includes
+#include "fileIO/Outputs/VTKOutput.hpp"
+#include "mainInterface/ProblemManager.hpp"
+
+#include "PyVTKOutputType.hpp"
+#include "dataRepository/python/PyGroupType.hpp"
+
+
+#define VERIFY_NON_NULL_SELF( self ) \
+  PYTHON_ERROR_IF( self == nullptr, PyExc_RuntimeError, "Passed a nullptr as self.", nullptr )
+
+#define VERIFY_INITIALIZED( self ) \
+  PYTHON_ERROR_IF( self->group == nullptr, PyExc_RuntimeError, "The PyVTKOutput is not initialized.", nullptr )
+
+namespace geosx
+{
+namespace python
+{
+
+struct PyVTKOutput
+{
+  PyObject_HEAD
+
+  static constexpr char const * docString =
+    "A Python interface to VTKOutput.";
+
+  geosx::VTKOutput * group;
+};
+
+
+static PyObject * PyVTKOutput_new( PyTypeObject *type, PyObject *args, PyObject *kwds )
+{
+  GEOSX_UNUSED_VAR( args, kwds );
+  PyVTKOutput *self;
+
+  self = (PyVTKOutput *)type->tp_alloc( type, 0 );
+  if( self != nullptr )
+  {
+    self->group = nullptr;
+  }
+
+  return (PyObject *)self;
+}
+
+
+static PyObject * PyVTKOutput_repr( PyObject * const obj ) noexcept
+{
+  PyVTKOutput const * const pyVTKOutput = LvArray::python::convert< PyVTKOutput >( obj, getPyVTKOutputType() );
+  if( pyVTKOutput == nullptr )
+  {
+    return nullptr;
+  }
+
+  VERIFY_INITIALIZED( pyVTKOutput );
+
+  string repr;
+  string const path = pyVTKOutput->group->getPath();
+  string const type = LvArray::system::demangle( typeid( *(pyVTKOutput->group) ).name() );
+  repr = path + " ( " + type + " )";
+
+  return PyUnicode_FromString( repr.c_str() );
+
+}
+
+
+static PyObject * output( PyVTKOutput * self, PyObject * args )
+{
+  VERIFY_NON_NULL_SELF( self );
+  VERIFY_INITIALIZED( self );
+
+  double time;
+  double dt;
+
+  if( !PyArg_ParseTuple( args, "dd", &time, &dt ) )
+  {
+    return nullptr;
+  }
+
+  geosx::ProblemManager & problem = self->group->getGroupByPath< ProblemManager >( "/Problem/" );
+  geosx::DomainPartition & domain = problem.getDomainPartition();
+
+  int cycleNumber = int(time/dt);
+  try
+  {
+    self->group->execute( time, dt, cycleNumber, 0, 0, domain );
+  }
+  catch( std::out_of_range const & e )
+  {
+    std::cout<<"Target not found. Impossible output."<<std::endl;
+  }
+  Py_RETURN_NONE;
+}
+
+static PyObject * setOutputDir( PyVTKOutput * self, PyObject * args )
+{
+  VERIFY_NON_NULL_SELF( self );
+  VERIFY_INITIALIZED( self );
+
+  PyObject * unicodePath;
+  // string fileNameRoot;
+
+  if( !PyArg_ParseTuple( args, "U", &unicodePath ) )
+  {
+    return nullptr;
+  }
+
+  LvArray::python::PyObjectRef<> asciiPath { PyUnicode_AsASCIIString( unicodePath ) };
+  if( asciiPath == nullptr )
+  {
+    return nullptr;
+  }
+
+  char const * const path = PyBytes_AsString( asciiPath );
+  if( path == nullptr )
+  {
+    return nullptr;
+  }
+
+  try
+  {
+    self->group->setOutputDirectory( path );
+    self->group->postProcessInput();
```

Comment:
I am not sure that calling `postProcessInput()` here is a good idea. I think it could break if you call `setOutputDir()` before calling `apply_initial_conditions()` on python side. Can you test it ?

Reply:
It seems to be OK even if I call `apply_initial_conditions()` after the `setOutputDir()`. No breaking and the name is correctly changed.

Reply:
Nice !